model-engine/model_engine_server/infra/services/image_cache_service.py

```diff
@@ -145,18 +144,18 @@ async def execute(self, endpoint_infra_states: Dict[str, Tuple[bool, ModelEndpoi
                     if state.resource_state.gpu_type == gpu_type and (
                         (
                             state.image not in images_to_cache_priority[key]
-                            or last_updated_at.replace(
-                                tzinfo=images_to_cache_priority[key][
-                                    state.image
-                                ].last_updated_at.tzinfo
-                            )
+                            or last_updated_at.replace(tzinfo=pytz.UTC)
                             > images_to_cache_priority[key][state.image].last_updated_at
                         )
                         and self.docker_repository.image_exists(image_tag, repository_name)
                     ):
                         images_to_cache_priority[key][state.image] = cache_priority
 
         images_to_cache = CachedImages(cpu=[], a10=[], a100=[], t4=[])
+        for key, val in images_to_cache_priority.items():
+            for image in images_to_cache_priority[key]:
+                images_to_cache_priority[key][image].last_updated_at.replace(tzinfo=pytz.UTC)
```

Comment:
does .replace update the original object?

Reply:
it doesn't look like it updates things inplace, changing it to be a new value assignment.